frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -300,24 +312,46 @@ export const tabsReducer = createReducer<DashboardState>(
 
     builder.addCase(
       moveDashCardToTab,
-      (state, { payload: { dashCardId, destTabId } }) => {
-        const { dashId } = getPrevDashAndTabs({ state });
-        if (dashId === null) {
-          throw Error(
-            `MOVE_DASHCARD_TO_TAB was dispatched but dashId (${dashId}) is null`,
-          );
-        }
+      (state, { payload: { dashCardId, destinationTabId } }) => {
         const dashCard = state.dashcards[dashCardId];
+        const dashboardId = checkNotNull(state.dashboardId);
+        const dashboard = state.dashboards[dashboardId];
+
+        dashboard.dashCardTabMovements ??= {};
+
+        dashboard.dashCardTabMovements[dashCardId] = {
```

Comment:
This is hard to manage / support. We should get all the information required for the undo action before moving the card and use it for undo. No global state changes should be required.